packages/grafana-data/src/transformations/transformers/formatString.ts

```diff
@@ -0,0 +1,90 @@
+import { map } from 'rxjs/operators';
+
+import { Field, FieldType } from '../../types';
+import { DataTransformerInfo } from '../../types/transformations';
+
+import { DataTransformerID } from './ids';
+
+export enum FormatStringOutput {
+  UpperCase = 'Upper Case',
+  LowerCase = 'Lower Case',
+  FirstLetter = 'Capitalize First Letter',
+  EveryFirstLetter = 'Capitalize First Letter Of Every Word',
+  PascalCase = 'PascalCase',
+  CamelCase = 'camelCase',
+}
+
+export interface FormatStringTransformerOptions {
+  stringField: string;
+  outputFormat: string;
+}
+
+export const formatStringTransformer: DataTransformerInfo<FormatStringTransformerOptions> = {
+  id: DataTransformerID.formatString,
+  name: 'Format string',
+  description: 'Set the capitalisation of a string field',
+  defaultOptions: { stringField: '', outputFormat: '' },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        const formatter = createStringFormatter(options.stringField, options.outputFormat);
+
+        if (!Array.isArray(data) || data.length === 0) {
+          return data;
+        }
+
+        return data.map((frame) => ({
+          ...frame,
+          fields: formatter(frame.fields),
+        }));
+      })
+    ),
+};
+
+/**
+ * @internal
+ */
+export const createStringFormatter = (stringField: string, outputFormat: string) => (fields: Field[]) => {
+  return fields.map((field) => {
+    // Find the configured field
+    if (field.name === stringField) {
+      // Update values to use the configured format
+      const newVals = field.values.map((value: String) => {
+        switch (outputFormat) {
+          case FormatStringOutput.UpperCase:
+            return value.toUpperCase();
+          case FormatStringOutput.LowerCase:
+            return value.toLowerCase();
+          case FormatStringOutput.FirstLetter:
+            return value.charAt(0).toUpperCase() + value.slice(1);
+          case FormatStringOutput.EveryFirstLetter:
+          case FormatStringOutput.PascalCase:
+          case FormatStringOutput.CamelCase:
```

Comment:
I'm personally not a fan of nested case statements. Is there a better way to do this?

Reply:
I think another option here would be something similar to what [`binaryOperators.ts`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/utils/binaryOperators.ts) does. I'm not really biased towards any in particular tho